rgbd/src/client.cpp

```diff
@@ -4,55 +4,134 @@
 
 #include "rgbd/image.h"
 #include "rgbd/ros/conversions.h"
+#include "rgbd/tools.h"
+
+#include <ros/console.h>
+#include <ros/duration.h>
+#include <ros/rate.h>
+
+#include <std_msgs/String.h>
+
 
 namespace rgbd {
 
 // ----------------------------------------------------------------------------------------
 
-Client::Client()
+Client::Client() : client_impl_mode_(ClientImplMode::rgbd)
 {
+    const std::string& hostname = get_hostname();
```

Comment:
First I checked if the return of `get_hostname` was empty. It that was the case, an runtime error would be thrown. This is now moved to `get_hostname`. So this can indeed be merged into one line.